merchant/src/catalog/backend.rs

```diff
@@ -158,6 +219,98 @@ impl CatalogService for CatalogSQLService {
         Ok(result.to_catalog_entry_document()?)
     }
 
+    async fn bulk_create(
+        &self,
+        account: Account,
+        catalog: Vec<CatalogObjectBulkDocument<String>>,
+    ) -> Result<Vec<SqlCatalogObjectDocument>, CatalogError> {
+        let mut objects_created: HashMap<String, SqlCatalogObjectDocument> = HashMap::new();
+        let mut objects_dependency: HashMap<String, u32> = HashMap::new();
+        let mut objects_dependency_document: HashMap<String, &CatalogObject<String>> =
+            HashMap::new();
+
+        // we count all references to the dependency
+
+        for (index, item) in catalog.iter().enumerate() {
+            let key_id = match &item.id {
+                Some(id) => make_id_by_alias(id),
+                None => make_id_by_index(index.try_into().unwrap()),
+            };
+
+            objects_dependency.entry(key_id.clone()).or_insert(0);
+            objects_dependency_document
+                .entry(key_id)
+                .or_insert(&item.catalog_object);
+
+            match &item.catalog_object {
+                CatalogObject::Variation(entry) => {
+                    let key = make_id_by_alias(&entry.item_id);
+                    objects_dependency.entry(key).and_modify(|e| *e += 1);
+                }
+                CatalogObject::Modification(entry) => {
+                    let key = make_id_by_alias(&entry.item_id);
+                    objects_dependency.entry(key).and_modify(|e| *e += 1);
+                }
+                _ => {}
+            }
+        }
+
+        let mut items_sorted_to_insert: Vec<(&String, &u32)> = objects_dependency.iter().collect();
+        items_sorted_to_insert.sort_by(|a, b| b.1.cmp(a.1));
+        // we start creating the dependencies from the less dependant
+        for (alias_id, _) in items_sorted_to_insert.clone() {
+            let item = *objects_dependency_document.get(alias_id).unwrap();
+            match item {
+                CatalogObject::Item(item) => {
+                    let catalog_object: CatalogObject<self::Id> =
+                        CatalogObject::<u32>::Item(item.clone());
+                    let document = self.create(account.clone(), &catalog_object).await?;
+                    objects_created.entry(alias_id.clone()).or_insert(document);
+                }
+                CatalogObject::Variation(variation) => {
```

Comment:
```suggestion
                ref variation @ CatalogObject::Variation(ref var) => {
```
You can bind the whole expression to a variable and use that instead since you are putting it back into the enum later with a clone anyway. The `ref`s here allow to also use the inner value